File: app/src/main/java/com/android/malsoftware/techadvisor/MillDetailValues.java
package com.android.malsoftware.techadvisor;

import android.annotation.SuppressLint;
import android.content.Context;

import java.util.ArrayList;
import java.util.List;

public class MillDetailValues {
    private Context mContext;
    private List<String> mFieldNames = new ArrayList<String>();
    private List<Pairs> mPairsArray = new ArrayList<>();

    private MillDetailValues(Context context) {
        mContext = context;
        FieldType[] val = FieldType.values();
        for (int i = 0; i < FieldType.values().length; ++i) {
            mPairsArray.add(new Pairs(val[i], 0));
            mFieldNames.add(String.valueOf(val[i]));
        }
    }

    @SuppressLint("StaticFieldLeak")
    private static MillDetailValues mMillDetailValues;

    public static MillDetailValues newInstance(Context context) {
        if (mMillDetailValues == null) mMillDetailValues = new MillDetailValues(context);
        return mMillDetailValues;
    }

    public List<Pairs> getPairsArray() {
        return mPairsArray;
    }

    public String getStringFieldValue(int position) {
        return String.valueOf(mPairsArray.get(position).getValue());
    }

    public List<String> getStringKeys() {
        return mFieldNames;
    }

    public FieldType getFieldType(int pos){
        return mPairsArray.get(pos).getFieldType();
    }
}

File: app/src/main/java/com/android/malsoftware/techadvisor/MillDetailValues.kt
package com.android.malsoftware.techadvisor

import android.annotation.SuppressLint
import android.content.Context

import java.util.ArrayList

class MillDetailValues private constructor(private val mContext: Context) {
    private val mFieldNames = ArrayList<String>()
    private val mPairsArray = ArrayList<Pairs>()

    val pairsArray: List<Pairs>
        get() = mPairsArray

    val stringKeys: List<String>
        get() = mFieldNames

    init {
        val `val` = FieldType.values()
        for (i in FieldType.values().indices) {
            mPairsArray.add(Pairs(`val`[i], 0))
            mFieldNames.add(`val`[i].toString())
        }
    }

    fun getStringFieldValue(position: Int): String {
        return mPairsArray[position].value.toString()
    }

    fun getFieldType(pos: Int): FieldType {
        return mPairsArray[pos].fieldType
    }

    companion object {

        @SuppressLint("StaticFieldLeak")
        private var mMillDetailValues: MillDetailValues? = null

        fun newInstance(context: Context?): MillDetailValues {
            if (mMillDetailValues == null) mMillDetailValues = context?.let { MillDetailValues(it) }
            return mMillDetailValues as MillDetailValues
        }
    }
}